workloads/logging.yml

```diff
@@ -44,57 +47,58 @@
-      when: LABAL_ALL_NODES
+      when: LABEL_ALL_NODES
       tags: label_node
     - name: Create workloads directory
       file:
-        path: "{{ansible_facts['env']['HOME']}}/workloads"
+        path: "{{WORKLOAD_DIR}}/workloads"
         state: directory
         mode: '0755'
     - name: Template cluster-loader logtest logging config file
       template:
         src: templates/logtest.yml.j2
-        dest: "{{ansible_facts['env']['HOME']}}/workloads/logtest.yml"
+        dest: "{{WORKLOAD_DIR}}/workloads/logtest.yml"
     - name: Template logtest-rc.json
       template:
         src: templates/logtest-rc.json.j2
-        dest: "{{ansible_facts['env']['HOME']}}/workloads/logtest-rc.json"
+        dest: "{{WORKLOAD_DIR}}/workloads/logtest-rc.json"
     - name: Podman pull cluster-loader image
       podman_image:
         name: quay.io/openshift/origin-tests
         tag: "{{ORIGIN_TESTS_VERSION}}"
     - name: Launch cluster-loader
       shell: |
         podman run --rm \
-        -v {{ ansible_facts['env']['KUBECONFIG']|default(ansible_facts['env']['HOME'] + '/.kube/config') }}:/root/.kube/config:z \
-        -v {{ansible_facts['env']['HOME']}}/workloads/:/root/workloads/:z \
+        -v {{ KUBECONFIG }}:/root/.kube/config:z \
+        -v {{WORKLOAD_DIR}}/workloads/:/root/workloads/:z \
         -i quay.io/openshift/origin-tests:{{ORIGIN_TESTS_VERSION}} \
         /bin/bash -c 'export KUBECONFIG=/root/.kube/config && \
         export VIPERCONFIG=/root/workloads/logtest.yml && \
         openshift-tests run-test "{{ CLUSTER_LOADER_STRING }} concurrently with templates [Suite:openshift]"' \
         |& tee workloads/cluster-loader.log
       tags: clusterloader
-    - name: Puase for time of test + PAUSE_OFFSET
+    - set_fact:
+        test_start: "{{ lookup('pipe','date +%s') }}"
+    - name: Pause for est. time of test "{{ (NUM_LINES|int / RATE|int) | round(0, 'floor') | int }}" min
       pause:
-        minutes: "{{ (NUM_LINES|int / RATE|int + PAUSE_OFFSET|int) | round(0, 'ceil') | int }}"
-      tags: clusterloader
-    - name: Get number of messages indexed
+        minutes: "{{ (NUM_LINES|int / RATE|int) | round(0, 'floor') | int }}"
+    - name: Check for test completion for up to "{{ PAUSE_OFFSET }}" min
       shell: |
-        BASE_CMD="oc exec -n openshift-logging -c elasticsearch {{ es_pod.stdout }} -- curl --connect-timeout 2 -s -k --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key"
-        ${BASE_CMD} 'https://localhost:9200/app*/_count' -H 'Content-Type: application/json' \
+        oc exec -n openshift-logging -c elasticsearch {{ es_pod.stdout }} -- es_util --query=app*/_count \
           -d '{"query":{"wildcard":{"kubernetes.namespace_name":{"value":"{{PROJECT_BASENAME}}*","boost":1,"rewrite":"constant_score"}}}}' | jq '.count'
       register: num_indexed
-    - debug:
-        var: num_indexed
-    - name: Assert number of messages is equal to NUM_LINES*NUM_PROJECTS
```

Comment:
This assertion shouldn't be removed from the task list because the playbook needs pass/fail output for automation purposes. I suggest reordering like so:
- Test Runtime
- Assert
- Cleanup

Reply:
Maybe I should modify here to register and assert.  The change here is an attempt to simply exit early.  There is no need to automatically pause when you may simply have satisfied the test seconds after the last message makes it to the store

Reply:
I definitely like that addition and giving more information about exactly how long the test took